plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergTableProperties.java

```diff
@@ -252,6 +297,48 @@ private static void validateFormatVersion(int version)
         }
     }
 
+    public static void validateCompression(IcebergFileFormat fileFormat, Optional<HiveCompressionCodec> compressionCodec, OptionalInt compressionLevel)
+    {
+        if (compressionCodec.isPresent()) {
+            if (!isCompressionCodecSupportedForFormat(fileFormat, compressionCodec.get())) {
+                throw new TrinoException(NOT_SUPPORTED, format("Compression codec LZ4 not supported for %s", fileFormat.humanName()));
```

Comment:
Error message here mentions `LZ4` explicitly, but it seems like this should be a parameter based on `compressionCodec.get()`

Reply:
done